tests/tracer/test_processors.py

```diff
@@ -216,3 +219,89 @@ def test_aggregator_partial_flush_2_spans():
     assert writer.pop() == [child1, child2]
     parent.finish()
     assert writer.pop() == [parent]
+
+
+def test_trace_tags_processor_applies_dd_origin_tags_to_each_span():
```

Comment:
These tests add too much dependence on implementation details which will make it painful to make refactors later on.

I think I'd prefer these tests as being integration tests in `test_tracer.py`. Maybe snapshotting would be handy to ensure each of the spans gets the origin tag?

Reply:
Agreed that these tests are very dependent on implementation details, I'll move them over to `test_tracer.py` as integration tests. I'm not sure about the benefit of snapshotting though, since all we really check (or even add) are origin tags to each span.

Reply:
☝🏻 yeah, it is a much nicer way than doing manual assertions, but I think we can handle that as a follow-up